### PR TITLE
Revert "TASK: Add type declarations in PersistenceManagerInterface"

### PR DESCRIPTION
**Pull request by mficzel**
This reverts commit 18386878d77b5d93ff976769f967a3713ce2d7e6. 

The return type `object` is not handled properly yet in proxy class building which causes trouble on the neos side. We are reverting this now as hotfix. The proper handling of `object` types can be added as bugfix. The type annotations for the PersistenceManagerInterface will have to wait until next major.

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)

### CODE REVIEW

**Review by albe — APPROVED** [review]

**Comment by mficzel** [comment]
Merging without second review because we need a green neos again

**Review by kdambekalns — COMMENTED** [review]
👍 post-merge - thanks for taking care.

**Pull request merged**
